src/ripple/nodestore/impl/Shard.cpp

```diff
@@ -424,29 +442,317 @@ Shard::valFetch(uint256 const& hash)
+                        "(TransID, Account, LedgerSeq, TxnSeq) VALUES ('");
+
+                    // Try to make an educated guess on how much space we'll
+                    // need for our arguments. In argument order we have:
+                    // 64 + 34 + 10 + 10 = 118 + 10 extra = 128 bytes
+                    sql.reserve(sql.length() + (accounts.size() * 128));
+
+                    for (auto const& account : accounts)
+                    {
+                        sql += sTxID;
+                        sql += "','";
+                        sql += ripple::toBase58(account);
+                        sql += "',";
+                        sql += sSeq;
+                        sql += ',';
+                        sql += sTxSeq;
+                        sql += "), ('";
+                    }
+
+                    sql.erase(sql.size() - 4);
+                    sql += ';';
+                    session << sql;
+
+                    JLOG(j_.trace()) <<
+                        "shard " << index_ << " account transaction: " << sql;
+                }
+                else
+                {
+                    JLOG(j_.warn()) <<
+                        "shard " << index_ <<
+                        " transaction in ledger " << sSeq <<
+                        " affects no accounts";
+                }
+
+                Serializer s;
+                item.second->add(s);
+                session <<
+                   (STTx::getMetaSQLInsertReplaceHeader() +
+                       item.first->getMetaSQL(
+                           seq,
+                           sqlEscape(std::move(s.modData())))
+                       + ';');
+            }
+        }
+
+        tr.commit ();
+    }
+
+    auto& session {lgrSQLiteDB_->getSession()};
+    soci::transaction tr(session);
+
+    session <<
+        "DELETE FROM Ledgers WHERE LedgerSeq = :seq;"
+        , soci::use(seq);
+    session <<
+        "INSERT OR REPLACE INTO Ledgers ("
+            "LedgerHash, LedgerSeq, PrevHash, TotalCoins, ClosingTime,"
+            "PrevClosingTime, CloseTimeRes, CloseFlags, AccountSetHash,"
+            "TransSetHash)"
+        "VALUES ("
+            ":ledgerHash, :ledgerSeq, :prevHash, :totalCoins, :closingTime,"
+            ":prevClosingTime, :closeTimeRes, :closeFlags, :accountSetHash,"
+            ":transSetHash);",
+        soci::use(to_string(ledger->info().hash)),
+        soci::use(seq),
+        soci::use(to_string(ledger->info().parentHash)),
+        soci::use(to_string(ledger->info().drops)),
+        soci::use(ledger->info().closeTime.time_since_epoch().count()),
+        soci::use(ledger->info().parentCloseTime.time_since_epoch().count()),
+        soci::use(ledger->info().closeTimeResolution.count()),
+        soci::use(ledger->info().closeFlags),
+        soci::use(to_string(ledger->info().accountHash)),
+        soci::use(to_string(ledger->info().txHash));
+
+    tr.commit();
+    return true;
```

Comment:
should this method also catch `soci_error` and return false in the case of such and exception?